internal/vendors/accessor.go

```diff
@@ -22,8 +22,22 @@ const (
 		VALUES 
 			(:id, :name, :email, :description, :bp_id, :bp_name, :rating, :area_group_id, :area_group_name, :sap_code, :modified_date, :modified_by, :dt)
 	`
-	getBulkByID          = `SELECT * FROM vendor WHERE id IN (?)`
-	getAllLocationsQuery = `SELECT DISTINCT area_group_name FROM vendor`
+	getBulkByID                 = `SELECT * FROM vendor WHERE id IN (?)`
+	getAllLocationsQuery        = `SELECT DISTINCT area_group_name FROM vendor`
+	getAllVendorIdByProductName = `
+		SELECT
+			v.id
+		FROM
+			vendor v
+		JOIN 
+			price pp ON pp.vendor_id = v.id
+		JOIN
+			product_vendor pv ON pv.id = pp.product_vendor_id
+		JOIN 
+			product p ON p.id = pv.product_id
+		WHERE
+			p.name = :product_name
```

Comment:
Isn't it too strict if we filter the vendor by **EXACT** product name, as it will cause user struggling to find the exact product name? Why don't we follow filter vendor by product name implementation?

Reply:
This is an API receiving request from a third party which is expected to be automated right. I'm assuming they will fetch the product name based on their data and we should have the same data. The case for filter vendors by product name is that users manually type the product name. I think since it is automated we don't want to send an email to a wrong vendor caused by an unintentionally mismatched. What do you think?